android/src/main/kotlin/com/icapps/background_location_tracker/flutter/FlutterBackgroundManager.kt

```diff
@@ -2,25 +2,28 @@ package com.icapps.background_location_tracker.flutter
 
 import android.content.Context
 import android.location.Location
+import android.os.Handler
+import android.os.Looper
 import com.icapps.background_location_tracker.BackgroundLocationTrackerPlugin
 import com.icapps.background_location_tracker.utils.Logger
 import com.icapps.background_location_tracker.utils.SharedPrefsUtil
 import io.flutter.embedding.engine.FlutterEngine
 import io.flutter.embedding.engine.dart.DartExecutor
+import io.flutter.embedding.engine.loader.FlutterLoader
 import io.flutter.embedding.engine.plugins.shim.ShimPluginRegistry
 import io.flutter.plugin.common.MethodCall
 import io.flutter.plugin.common.MethodChannel
 import io.flutter.view.FlutterCallbackInformation
-import io.flutter.view.FlutterMain
 
 internal object FlutterBackgroundManager {
     private const val BACKGROUND_CHANNEL_NAME = "com.icapps.background_location_tracker/background_channel"
 
+    private val flutterLoader = FlutterLoader()
+
     private fun getInitializedFlutterEngine(ctx: Context): FlutterEngine {
         Logger.debug("BackgroundManager", "Creating new engine")
 
         val engine = FlutterEngine(ctx)
-        FlutterMain.ensureInitializationComplete(ctx, null)
         //Backwards compatibility with v1. We register all the user's plugins.
```

Comment:
Should we still retain this compatibility?

Reply:
Other plugins still have this support. So I would still keep this. (If not remove it in another pr with another build)